lib/conjur/dsl2/doc.rb

```diff
@@ -0,0 +1,43 @@
+module Conjur
+  module DSL2
+    module Doc
+      Attribute = Struct.new(:id, :kind)
+
+      Operation = Struct.new(:id, :super_id, :description, :example, :attributes)
+      
+      class << self
+        def list
+          all_types = Set.new
+          new_types = Set.new
+          new_types += Conjur::DSL2::Types::Base.subclasses
+          all_types += new_types
+          while !new_types.empty?
+            iteration_new_types = Set.new
+            new_types.each do |type|
+              subtypes = type.subclasses
+              iteration_new_types += (Set.new(subtypes) - all_types)
+              all_types += subtypes
+            end
+            new_types = iteration_new_types.dup
+            iteration_new_types.clear
+          end
+          all_types.map do |type|
+            # TODO: I am not sure what this is
+            next if type == Conjur::DSL2::Ruby::Policy
+
+            description = type.send(:description) rescue ""
+            example = type.send(:example) rescue ""
```

Comment:
The example should be parsed as YAML to make sure that it's syntactically valid.